fv3core/utils/grid.py

```diff
@@ -457,7 +465,7 @@ def max_shape(self):
         This should rarely be required, consider using appropriate calls to helper
         methods that get the correct shape for your particular variable.
         """
-        return self.domain_full(add=(1, 1, 1))
+        return self.domain_full(add=(1, 1, 1 + self.origin[2]))
```

Comment:
This is a little confusing, this is to account for that you might be vertically restricting  the domain, but then could still get out the full max_shape? I think it probably is needed

Reply:
Discussed and resolved.

Reply:
Yeah it's definitely confusing. Added a comment to avoid future confusion.